.github/workflows/unit_test.yml

```diff
@@ -0,0 +1,49 @@
+# This workflow will build a Java project with Gradle
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-gradle
+
+name: Run Unit Tests
+
+on: [push]
+
+jobs:
+  unit-test:
+
+    runs-on: ubuntu-latest
+
+    services:
+      postgres:
+        image: postgres:12.3
+        env:
+          POSTGRES_PASSWORD: postgres
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          - 5432:5432
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Initialize Postgres DB
+        env:
+          PGPASSWORD: postgres
+        run: psql -h 127.0.0.1 -U postgres -f ./local-dev/local-postgres-init.sql
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - name: Cache Gradle packages
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: v1-${{ runner.os }}-gradle-${{ hashfiles('**/gradle-wrapper.properties') }}-${{ hashFiles('**/*.gradle') }}
+          restore-keys: v1-${{ runner.os }}-gradle-${{ hashfiles('**/gradle-wrapper.properties') }}
+      - name: Grant execute permission for gradlew
+        run: chmod +x gradlew
+      - name: Run unit tests
+        run: ./gradlew unitTest --scan
+      - name: Stop Postgres DB
+        run:  ./local-dev/run_postgres.sh stop
```

Comment:
do we need to stop the database? Maybe keeping this just to test that `run_postgres.sh stop` works is a good reason to have it.
if so, we probably still want it to run even if the unit tests fail, as long as the initialize database step has succeeded? see https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsif

Reply:
ah, sorry, forgot to delete this line. That is my previous version: Manually start and stop ad postgres in docker. 
Now we are using postgres service container (line 13-24) so we don't need to manually start and stop it.